file/sst_file_manager_impl.cc

```diff
@@ -80,6 +80,23 @@ Status SstFileManagerImpl::OnAddFile(const std::string& file_path,
   return Status::OK();
 }
 
+Status SstFileManagerImpl::OnAddFileForDestroyDB(const std::string& file_path,
+                                                 uint64_t* file_size) {
+  uint64_t num_hard_links = 1;
+  Status s =
+      fs_->NumFileLinks(file_path, IOOptions(), &num_hard_links, nullptr);
+  if (s.ok() && num_hard_links > 1) {
+    *file_size = 0;
+  } else {
+    s = fs_->GetFileSize(file_path, IOOptions(), file_size, nullptr);
+  }
+  if (s.ok()) {
+    MutexLock l(&mu_);
+    OnAddFileImpl(file_path, *file_size);
```

Comment:
Why do we need to "track" the file?

Reply:
This is to support `MyRocks` use case where they are mostly using `DestroyDB` to remove obsolete checkpoints, not the original DB. They are passing a new `SstFileManager` for this purpose https://github.com/facebook/mysql-5.6/blob/9d0a754dc9973af0508b3ba260fc337190a3218f/storage/rocksdb/ha_rocksdb.cc#L1284

The files are untracked by this `SstFileManager` so I added this tracking, to satisfy the `DeleteScheduler`'s slow deletion logic.

Reply:
OK, now I see how DeleteScheduler depends on SstFileManager tracking. (Trash size in delete scheduler vs. total tracked size in SstFileManager.) Weird.

I think it would make more sense to put these files into the deletion pipeline of the shared manager/scheduler, usually subject to `rate_bytes_per_sec`, but without counting them toward trash size and associated ratio.

Reply:
That's a good point. Let me check with MyRocks. The part that special case files with more than 1 hard links and immediately delete them are still useful for checkpoint destroy. As most of the files will fall into that category, slowly deleting them would be too slow. 

Any concerns if I directly add this functionality to `DeleteDBFile` and do this for both `DestroyDB` and for when DB is running normally?

Reply:
Thanks for the suggestions! I have checked with MyRocks about this workaround. They will do some testing and let's not proceed with this change before they get back to us.

Reply:
Regarding not tracking files given to DestroyDB:
Perhaps ideally the size tracking would include the size of all DBs (open or not) on your storage. At first glance, when we have an opportunity to include closed DBs in the tracking, that would seem to given a closer representation of the "true" trash ratio (if we knew about all the closed DBs sitting around). But since all the files given to DestroyDB are considered trash, we have selection bias. The “true” trash ratio is misrepresented if the only files represented among closed DBs are those that are known to be trash.

Reply:
I have updated the PR to support deleting untracked files and use it for DestroyDB. Let me know if this looks OK.